packages/driver/src/util/config_utilities.ts

```diff
@@ -0,0 +1,95 @@
+import _ from 'lodash'
+import structuredClonePonyfill from 'core-js-pure/actual/structured-clone'
+import { options } from '@packages/config'
+
+const UNSERIALIZABLE = '__unserializable'
+
+// If a native structuredClone exists, use that to determine if a value can be serialized or not. Otherwise, use the ponyfill.
+// we need this because some implementations of SCA treat certain values as unserializable (ex: Error is serializable in ponyfill but NOT in firefox implementations)
+// @ts-ignore
+const structuredCloneRef = window?.structuredClone || structuredClonePonyfill
+
+/**
+ * Mutates the config/env object serialized from the other domain to omit read-only values
+ * to prevent trying to update differences in read-only config. This function does mutate this
+ * serialized config/env by reference, which should be ok since this object is a clone
+ * of what is being received back from the 'other' domain.
+ *
+ * @param objectLikeConfig the config/env object to omit read-only values from.
+ * @returns a reference to the config/env object passed in
+ */
+const omitReadOnlyDifferences = (objectLikeConfig: Cypress.ObjectLike) => {
+  Object.keys(objectLikeConfig).forEach((key) => {
+    if (options.find((option) => option.name === key)?.canUpdateDuringTestTime === false) {
+      delete objectLikeConfig[key]
+    }
+  })
+
+  return objectLikeConfig
+}
+
+/**
+ * Sanitizes any unserializable values from the config/env object to prep the config/env for postMessage serialization
+ * @param objectToSanitize Object that is either Cypress.config() or Cypress.env()
+ * @returns a copy of this object with all unserializable keys omitted from the object.
+ *
+ * NOTE: If an object contains an unserializable property, the whole object is deemed as unserializable
+ */
+const preprocessConfigOrEnvObj = <T>(objectToSanitize: { [key: string]: any }): T => {
```

Comment:
this function can be simplified to only iterate through the object once. We also might want to clone it with `clone` and omit unserialized properties to preserve the prototypical hierarchy when it matters

Reply:
nevermind. Looks like Objects lose prototypical inheritance over `postMessage`